src/internal/observe/observe.go

```diff
@@ -469,7 +431,26 @@ func waitAndCloseBar(bar *mpb.Bar, log func()) func() {
 // other funcs
 // ---------------------------------------------------------------------------
 
-const Bullet = "∙"
+var nop = func() {}
+
+// listen runs a
+func listen(ctx context.Context, ci <-chan struct{}, onEnd, onInc func()) {
```

Comment:
RE: my comment about a regression test. I think we should at minimum add a unit test for this. Kudos for pulling this logic out - makes it testable!

Reply:
done, tests have been added